did/ion/did.go

```diff
@@ -85,3 +108,216 @@ func LongToShortFormDID(longFormDID string) (string, error) {
 	}
 	return shortFormDID, nil
 }
+
+// PatchesToDIDDocument applies a list of sidetree state patches in order resulting in a DID Document.
+func PatchesToDIDDocument(shortFormDID, longFormDID string, patches []any) (*did.Document, error) {
```

Comment:
I strongly discourage the use of `any` unless there is a strong reason to do so. Can you clarify why it's preferable to have patches be `[]any`? 

An alternative is to define an interface that all patches implement.

```go
type Patch interface{
  isPatch()
}

func (a AddServicesAction) isPatch() { }
```

Reply:
will switch to interface